middleware/responder/sse/responder.go

```diff
@@ -0,0 +1,69 @@
+package sse
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/olebedev/emitter"
+	"github.com/siemens/wfx/middleware/logging"
+)
+
+// Responder sends a stream of events to the client.
+func Responder(ctx context.Context, events <-chan emitter.Event) middleware.ResponderFunc {
+	return func(w http.ResponseWriter, p runtime.Producer) {
+		log := logging.LoggerFromCtx(ctx)
+
+		flusher, _ := w.(http.Flusher)
+
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+
+		flusher.Flush()
+
+		id := 1
+	Loop:
+		for {
+			log.Debug().Msg("Waiting for next event")
+			select {
+			case ev, ok := <-events:
+				if !ok {
+					log.Debug().Msg("SSE channel is closed")
+					break Loop
+				}
+				b, err := json.Marshal(ev.Args[0])
+				if err != nil {
+					log.Error().Err(err).Msg("Failed to marshal status event")
+					continue Loop
+				}
+				log.Debug().RawJSON("event", b).Msg("Pushing SSE event to client")
+
+				_, _ = w.Write([]byte("data: "))
+				_, _ = w.Write(b)
+				_, _ = w.Write([]byte(fmt.Sprintf("\nid: %d\n\n", id))) // must end with two newlines
+
+				flusher.Flush()
+
+				id++
```

Comment:
Hm, what happens on overflow?

Reply:
Hm could happen yes. I have changed it to uint64, so it will wrap around to zero. This is unlikely to happen and the client can recognize this if it sees id 0.

Reply:
OK, as it's initially `1` a client can detect overflows when it reads `0`. Should be documented :)

Reply:
Added to documentation